controllers/users_controller.js

```diff
@@ -76,160 +49,116 @@ module.exports.findUser = async ({params: {username = null}}, res) => {
+module.exports.logOffAllOtherSessions = async (req, res, next) => {
   try {
     await userModel.logOffAllOtherSessions(req.user.id, req.user.uuid)
-    return res.status(200).json(
-      new Response('Successfully logged out of your other sessions')
-    )
+    return Response.OK(res, 'Successfully logged out of your other sessions')
   } catch (e) {
-    console.error(e)
-    return res.status(500).json(new ErrorResponse(
-      [new Error('An error occurred logging out of your other sessions')]
-    ))
+    next(e)
   }
 }
 
-module.exports.delete = async ({body: {password = null}, user = null}, res) => {
+module.exports.delete = async ({body: {password = null}, user = null}, res, next) => {
   const passwordMatch = await userModel.comparePasswordByID(user.id, password)
   if (!passwordMatch) {
-    return res.status(403).json(new ErrorResponse(
-      [new Error('Invalid password')]
-    ))
+    next(FSError.invalidPassword())
   }
 
   try {
     await userModel.delete(user.id)
-    return res.status(200).json(new Response('Successfully deleted user'))
+    return Response.OK(res, 'Successfully deleted user')
   } catch (e) {
-    console.error(e)
-    return res.status(500).json(new ErrorResponse(
-      [new Error('An error occurred trying to remove your account')]
-    ))
+    next(e)
   }
 }
 
-module.exports.updateProfilePicture = async (
-  {user: {id = null}, body: {image_url: imageURL = null}}, res) => {
+module.exports.updateProfilePicture = async ({user: {id = null}, body: {image_url: imageURL = null}}, res, next) => {
   try {
     await userModel.updateProfilePicture(id, imageURL)
-    return res.status(200).json(
-      new Response('Successfully updated your user profile')
-    )
+    return Response.OK(res, 'Successfully updated your user profile')
   } catch (e) {
-    console.error(e)
-    return res.status(500).json(new ErrorResponse(
-      [new Error('Failed to update user profile')]
-    ))
+    next(e)
   }
 }
 
-module.exports.updateBackgroundPicture = async (
-  {user: {id = null}, body: {image_url: imageURL = null}}, res) => {
+module.exports.updateBackgroundPicture = async ({user: {id = null}, body: {image_url: imageURL = null}}, res, next) => {
   try {
     await userModel.updateBackgroundPicture(id, imageURL)
-    return res.status(200).json(
-      new Response('Successfully updated your user profile')
-    )
+    return Response.OK(res, 'Successfully updated your user profile')
+  } catch (e) {
+    next(e)
+  }
+}
+
+module.exports.getDefaultProfilePicture = async ({user: {id = null}}, res, next) => {
+  try {
+    const [{username, datecreated}] = await userModel.findUserByID(id)
+    const today = new Date()
+    res.set({
+      'Content-Type': 'image/png',
+      'Cache-Control': `max-age=${60 * 60 * 24 * 365}`,
+      'Expires': new Date(today.getFullYear() + 1, today.getMonth(), today.getDate()),
+      'Last-Modified': datecreated
+    }).send(jdenticon.toPng(username, 500))
```

Comment:
@jonathanGB did I do the caching headers correctly?

Reply:
So this will generate a picture of size 500px * 500px?

Reply:
Yep